apps/emqx/src/emqx_frame.erl

```diff
@@ -166,8 +171,21 @@ parse_complete(
         <<0:8>> ->
             parse_bodyless_packet(Header);
         _ ->
-            {_RemLen, Rest2} = parse_variable_byte_integer(Rest1),
-            parse_packet_complete(Rest2, Header, Options)
+            {RemLen, Rest2} = parse_variable_byte_integer(Rest1),
+            case RemLen > byte_size(Rest2) of
+                true ->
+                    %% inet delivers an incomplete packet which means it's too large
+                    %% according to the [{packet_size, MaxPacketSize}] set to listener
+                    HeaderLen = byte_size(MaybeComplete) - byte_size(Rest2),
+                    FrameLen = HeaderLen + RemLen,
+                    ?PARSE_ERR(#{
```

Comment:
Maybe need to handle `frame_error` in [`emqx_connection`](https://github.com/thalesmg/emqx/blob/a1f9a0934f8b9bc3fd19f7d1446fb23b107731bf/apps/emqx/src/emqx_connection.erl#L754-L772) for [FVT to pass](https://github.com/emqx/emqx/actions/runs/18018233744/job/51272155057?pr=16024#step:6:174)?